### PR TITLE
Implement dismiss single notification request

### LINKED ISSUE #295: Implement Notifications API
Mastodon's Notifications API allows a user to receive notifications for activity on his account or statuses. The goal of this task is to implement the following endpoints of this API in BigBone:

- Get a single notification
- Dismiss a single notification

The Notifications API as it is documented by Mastodon can be found here:
https://docs.joinmastodon.org/methods/notifications/

Tip: A lot of other endpoints similar to this have already been implemented. In order to get an idea how to implement this, I suggest that you study the code upfront. Start with one of the `*Methods.kt` files.

Please consult the following documents to make sure your contribution meets our projects criteria:

- [Code of Conduct](https://github.com/andregasser/bigbone/blob/master/CODE_OF_CONDUCT.md)
- [Code Contribution Guidelines](https://github.com/andregasser/bigbone/blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by PattaFeuFeu**
# Description

Adds missing method to dismiss a single notification to the already existing notifications endpoint and updates endpoints that no longer matched those described in the documentation.

I renamed the existing methods `getNotifications` and `clearNotifications` to `getAllNotifications` and `dismissAllNotifications` to clarify the distinction between a single notification and all notifications and to get a bit closer to the headlines used in the documentation (not necessarily the API endpoints).
I’m open for this to be challenged as I’m not sure myself.

Closes #295 

# Type of Change

(Keep the one that applies, remove the rest)

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - Function name change
- This change requires a documentation update
  - [API Coverage](https://github.com/andregasser/bigbone/wiki/Mastodon-API-Coverage) page needs to be updated

# How Has This Been Tested?

`gradle check` with new unit tests added

# Mandatory Checklist

- [x] My change follows the projects coding style
- [x] I ran `gradle check` and there were no errors reported
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# Optional Things To Check

- [x] In case you worked on new features: Did you also do the Rx part?
- [x] In case you worked on endpoints: Please mention in the PR that [API Coverage](https://github.com/andregasser/bigbone/wiki/Mastodon-API-Coverage) page needs to be updated

### CODE REVIEW

File: bigbone/src/main/kotlin/social/bigbone/api/method/NotificationMethods.kt [hunk 1]
```diff
@@ -13,16 +13,21 @@ import social.bigbone.api.exception.BigBoneRequestException
  */
 class NotificationMethods(private val client: MastodonClient) {
```
Comment (bocops) [inline bigbone/src/main/kotlin/social/bigbone/api/method/NotificationMethods.kt:14]: Kdoc for the class references an alternative "api/vX/notification" (singular) endpoint, which was removed by code changes and apparently never existed, assuming a Mastodon doc error at the time of first implementing this. Please remove that reference.

**Review by bocops — CHANGES_REQUESTED** [review]
One small Kdoc change necessary. Rest looks fine for me, as does the suggested method renaming.

File: bigbone/src/main/kotlin/social/bigbone/api/method/NotificationMethods.kt (hunk 1 above)
Comment (PattaFeuFeu) [reply bigbone/src/main/kotlin/social/bigbone/api/method/NotificationMethods.kt:14]: Done in c5d3329

**Review by bocops — APPROVED** [review]
All good! :)

**Pull request merged**